src/frontend/src/expr/type_inference/mod.rs

```diff
@@ -95,6 +97,29 @@ impl From<DataType> for DataTypeName {
     }
 }
 
+impl From<DataTypeName> for AstDataType {
+    fn from(type_name: DataTypeName) -> Self {
+        match type_name {
+            DataTypeName::Boolean => AstDataType::Boolean,
+            DataTypeName::Int16 => AstDataType::SmallInt(None),
+            DataTypeName::Int32 => AstDataType::Int(None),
+            DataTypeName::Int64 => AstDataType::BigInt(None),
+            DataTypeName::Decimal => AstDataType::Decimal(None, None),
+            DataTypeName::Float32 => AstDataType::Real,
+            DataTypeName::Float64 => AstDataType::Double,
+            DataTypeName::Varchar => AstDataType::Varchar,
+            DataTypeName::Date => AstDataType::Date,
+            DataTypeName::Timestamp => AstDataType::Timestamp(false),
+            DataTypeName::Timestampz => AstDataType::Timestamp(true),
+            DataTypeName::Time => AstDataType::Time(false),
+            DataTypeName::Interval => AstDataType::Interval,
+            DataTypeName::Struct | DataTypeName::List => {
+                panic!("Functions returning struct or list can not be inferred. Please use `FunctionCall::new_unchecked`.")
```

Comment:
This error message is incorrect in the context.

Reply:
Moved to sqlsmith utils instead, returning `None` for invalid cases.
Caller has context (e.g. if return type cannot be casted), and hence it will handle `None` case.